add mdx provider init  doc (#8)
miamtech/mealz-documentation: docs/shared/overview/providerInit.mdx

import IosProviderId from '@site/docs/ios/advanced/_shared/_miamManager/ProviderId.md';
import IosBasicImp from '@site/docs/ios/overview/_shared/_MealzBasicImpProvider.mdx'
import IosUserSetting from '@site/docs/ios/overview/_shared/_MealzUserSetUp.mdx'
import IosStoreSetup from '@site/docs/ios/overview/_shared/_MealzStoreSetupProvider.mdx'

export function ProviderId(props){
    if(props.platform == "android") { return <AndroidProviderId /> } else return  <IosProviderId />
}

export function BasicImp(props){
    if(props.platform == "android") { return <AndroidBasicImp /> } else return  <IosBasicImp />
}

export function UserSetting(props){
    if(props.platform == "android") { return <AndroidUserSetting /> } else return  <IosUserSetting />
}

export function StoreSetup(props){
    if(props.platform == "android") { return <AndroidStoreSetup /> } else return  <IosStoreSetup />
}

You are providing content you use Mealz as a link to retailers

## ProviderKey - base64 key to set information

<ProviderId platform= {props.platform}/>

:::tip
Your keys will be provided by our Development team
:::

## Basic implementation

> There is several options to configure that handle authless here we are considering a basic implemtation
> were Mealz initialization process will only start after the user is **logged in** and has **selected a valid

Reviewer: ```suggestion
There are several options to configure to handle authless users. Here is a basic implemtation
where the Mealz initialization process will only start after the user is **logged in** and has **selected a valid store**.
```

> First to init Mealz we need your application context and also your provider key
> 
> :::note
> Two key must have been send to you one for **developement** purpose and a **production** one
> :::

Reviewer: ```suggestion
Before initializing Mealz we need your application context and also your provider key.

:::note
Two keys have been sent to you one for **developement** and one for **production**.
:::
```

<BasicImp platform ={props.platform}/>

## User setup

Here is how to pass the user ID to the SDK, directly within the host app:

:::info

> You can also enable **authless** feature, more informations [**here**](../advanced/user-configuration/##AuthLess

Reviewer: ```suggestion
You can also enable **authless** feature, more information [**here**](../advanced/user-configuration/##AuthLess
```
